src/lib/core/CHIPConfig.h

```diff
@@ -2449,6 +2449,15 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
 #define CHIP_IM_MAX_NUM_WRITE_CLIENT 4
 #endif
 
+/**
+ * @def CHIP_DEVICE_CONTROLLER_SUBSCRIPTION_ATTRIBUTE_PATH_POOL_SIZE
+ *
+ * @brief Defines the object pool for allocating attribute path for subscription in device controller.
+ */
+#ifndef CHIP_DEVICE_CONTROLLER_SUBSCRIPTION_ATTRIBUTE_PATH_POOL_SIZE
+#define CHIP_DEVICE_CONTROLLER_SUBSCRIPTION_ATTRIBUTE_PATH_POOL_SIZE CHIP_IM_MAX_NUM_READ_CLIENT
```

Comment:
For some applications, this is going to feel really restrictive.

Most of the metadata for a subscription is pretty lightweight.  What in the ReadClient is so large that we've decided we can only have 4 at a given time?  Also, is there a plan to provide a dynamic pool in applications where that makes sense?

Reply:
yes, agree. we would use object pool for several pool in interaction model engine in follow-up next week. for this one, we would also use object pool as well, file ticket https://github.com/project-chip/connectedhomeip/issues/9602

Reply:
As this lives in `CHIPConfig.h`, this is designed to be a default value.
There will be follow ups for dynamic pool in applications (https://github.com/project-chip/connectedhomeip/issues/9602).